pkg/postgres/pgutils/retry.go

```diff
@@ -0,0 +1,54 @@
+package pgutils
+
+import (
+	"time"
+
+	"github.com/stackrox/rox/pkg/timeutil"
+)
+
+const (
+	interval = 5 * time.Second
+	timeout  = 5 * time.Minute
+)
+
+// RetryExecQuery is used to specify how long to retry to successfully run an exec query
+// that fails with Transient errors
+func RetryExecQuery(fn func() error) error {
+	// Shape fn to match the RetryQuery below
+	fnWithReturn := func() (struct{}, error) {
+		return struct{}{}, fn()
+	}
+	_, err := RetryQuery(fnWithReturn)
+	return err
+}
+
+// RetryQuery is used to specify how long to retry to successfully run a query
+// that fails with Transient errors
+func RetryQuery[T any](fn func() (T, error)) (T, error) {
+	// Run query immediately
+	if T, err := fn(); err == nil || !isTransientError(err) {
+		return T, err
+	}
+
+	expirationTimer := time.NewTimer(timeout)
+	defer timeutil.StopTimer(expirationTimer)
+
+	intervalTicker := time.NewTicker(interval)
+	defer intervalTicker.Stop()
+
+	var err error
+	for {
+		select {
+		case <-expirationTimer.C:
+			log.Fatalf("unsuccessful in reconnecting to the database: %v. Exiting...", err)
```

Comment:
It might make sense, but it's a bit odd that a repeated failure of a transient error is _worse_ than a permanent error. I guess that a permanent error is assumed to mean that there's something wrong with the input?

Reply:
New logic basically says all errors are transient unless they are specifically postgres errors that are not transient in which case crashing won't do anything